src/main/java/com/bookstore/bookstoreapi/data/entity/Rating.java

```diff
@@ -0,0 +1,48 @@
+package com.bookstore.bookstoreapi.data.entity;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.ManyToOne;
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.Min;
+import jakarta.persistence.Table;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Entity
+@Table(name = "Rating")
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Rating {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "RatingID")
+    private Long id;
+    
+    @Column(name = "Rating")
+    @Min(1)
+    @Max(5)
+    private Integer rating;
+
+    @Column(name = "Book")
+    @ManyToOne
+    private Book book;
+
+    @Column
+    @ManyToOne
+    private User user;

```

Comment:
One thing I did not notice yesterday was this ^, what entity class is `User` class that you are calling, from what I can see in your changes you do not have any entity class called `User`.

Reply:
On top of this, you will need to establish `foreign key` relationship with the customer entity, `Rating` is the owner, and `Customer` is the minority owner.